utils/validators/service.js

```diff
@@ -0,0 +1,136 @@
+import isEmpty from 'lodash/isEmpty';
+import { validateDnsLabel, validateHostname } from '@/utils/validators';
+
+export function servicePort(spec, getters, errors, validatorArgs) {
+  const { ports, type: serviceType } = spec;
+
+  if (serviceType === 'ExternalName') {
+    return errors;
+  }
+
+  if (isEmpty(ports)) {
+    errors.push(getters['i18n/t']('validation.required', { key: 'Port Rules' }));
+
+    return errors;
+  }
+
+  ports.forEach((port, ind, ary) => {
+    const {
+      name,
+      nodePort,
+      port: pPort,
+      targetPort,
+    } = port;
+    const idx = ind + 1;
+
+    if (ary.length > 1 && isEmpty(name)) {
+      errors.push(getters['i18n/t']('validation.service.ports.name.required', { position: idx }));
+    }
+
+    if (!isEmpty(name)) {
+      const nameErrors = validateDnsLabel(name, 'name', getters, undefined, errors);
+
+      if (!isEmpty(nameErrors)) {
+        if (errors.length && errors.length > 0) {
+          errors = [...errors, ...nameErrors];
+        } else {
+          errors = nameErrors;
+        }
+      }
+    }
+
+    if (nodePort) {
+      const np = parseInt(nodePort, 10);
+
+      if (isNaN(np)) {
+        errors.push(getters['i18n/t']('validation.service.ports.nodePort.requiredInt', { position: idx }));
+      }
+    }
+
+    if (pPort) {
+      const p = parseInt(pPort, 10);
+
+      if (isNaN(p)) {
+        errors.push(getters['i18n/t']('validation.service.ports.port.requiredInt', { position: idx }));
+      }
+    } else {
+      errors.push(getters['i18n/t']('validation.service.ports.port.required', { position: idx }));
+    }
+
+    if (targetPort) {
+      const tp = parseInt(targetPort, 10);
+
+      if (isNaN(tp)) {
```

Comment:
This will break on services that start with a number (e.g. `parseInt("9pfs") === 9`.  Check for characters other than a number instead of if it parses as a number.

Reply:
Updated the code